### PR TITLE
Feature/relevant dataset extraction

### PR DESCRIPTION
**Pull request by samuelgoodall**
-review me senpai uwu :3

-changed original parserbot to now extract the relevant dataset and copy it to a new locating extracted
-added author_title_tuple_extractor to extract author title tuples
-added UsablePaperExtractor 
-trimmed down information extractor now only extracts information again -> removed bibitem parsing

### CODE REVIEW

File: bibitem_parsing/bibitem_parser.py [hunk 1]
```diff
@@ -98,7 +103,8 @@ def _strip_dangling_brackets_and_comma(self, unclean_string) -> str:
             if len(unclean_string) <= 2:
                 return unclean_string
             unclean_string = unclean_string[0:-1]
```
Comment (DJE98) [inline bibitem_parsing/bibitem_parser.py:105]: Add maybe a comment

File: bibitem_parsing/bibitem_parser.py [hunk 2]
```diff
@@ -176,7 +182,7 @@ def _parse_bibentrys_manually(self, tex_input_file: str) -> list:
             pass
```
Comment (DJE98) [inline bibitem_parsing/bibitem_parser.py]: Delete it

File: bibitem_parsing/bibitem_parser.py [hunk 3]
```diff
@@ -204,23 +210,29 @@ def convert_texfile_2_author_title_tuples(self, tex_input_file, algorithm: Algor
...
+        # dataitem = self._strip_special_chars(dataitem)
+        #              labels = self.neural_parscit.predict_for_text(dataitem,show=False)
+        #             author_title_tuples.append(self._convert_neural_parscit_output_too_author_title_tuple(labels, dataitem))
+        try:
+            data = self._parse_bibentrys_manually(tex_input_file=tex_input_file)
+            zipped_list = list(zip(author_title_tuples, data))
+            return zipped_list
+        except UnicodeDecodeError:
+            sys.stderr.write("Error message: Contains none unicode characters.\n")
+            pass
```
Comment (DJE98) [inline bibitem_parsing/bibitem_parser.py]: Delete pass

File: bibitem_parsing/bibitem_parser.py [hunk 4]
```diff
@@ -254,9 +266,8 @@ def check_how_many_titles_are_usable(self, author_title_tuple_list: list):
 
 if __name__ == "__main__":
     tex_input_file = 'symperC.bbl'  # '/mnt/c/Users/sgoodall/Desktop/archive/NLPProjekt/bibitem_parsing/tex2bib-master/example-cites.tex'
```
Comment (DJE98) [inline bibitem_parsing/bibitem_parser.py]: Delete main, you have the tests to test the functionality

File: metadata/README.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+Sqlite DB viewer: https://inloop.github.io/sqlite-viewer/
+
+the convertjsontodatabase script converts the metadata to a sqlite database.
+1. download the metadata (https://www.kaggle.com/datasets/Cornell-University/arxiv?resource=download) and place it into the metadata folder as archive.zip
+2. set database_name to the name of the database
+3. execute
+4. twiddle your thumbs
```
Comment (DJE98) [inline metadata/README.md:7]: Nice

File: metadata/convertjsontodatabase.py [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    :param file_path:
+    :return:
+    """
+    print("starting extraction")
+    with ZipFile(file_path, 'r') as file:
+        file.extractall()
+    print("finished extraction")
+
+
+if __name__ == '__main__':
```
Comment (DJE98) [inline metadata/convertjsontodatabase.py:98]: Extract main into a separate file

File: title_tuple_metadata_checker/authortitletuple.json [hunk 7]
```diff
@@ -0,0 +1,23 @@
+[
```
Comment (DJE98) [inline title_tuple_metadata_checker/authortitletuple.json]: Delete the whole file

File: usable_dataset_tar_extractor.py [hunk 8]
```diff
@@ -36,25 +38,22 @@ def run(self):
...
             with open("save.json", "w") as backup_file:
                 backup_file.writelines(json.dumps(data, indent=7))
 
 
 if __name__ == "__main__":
-    parser = ParserBot("arXiv_src_1705_015.tar", "content/", "extract/")
+    parser = UsableDatasetTarExtractor(None, "content/", "extract/", "usable_dataset/")
+
     print("Parser started")
     parser.run()
```
Comment (DJE98) [inline usable_dataset_tar_extractor.py:59]: Missing Class for run through the extracted paper.

File: metadata/convertjsontodatabase.py (hunk 6 above)
Comment (samuelgoodall) [reply metadata/convertjsontodatabase.py:98]: ne würd so lassen skript macht nur das nicht mehr nicht weniger

File: usable_dataset_tar_extractor.py (hunk 8 above)
Comment (samuelgoodall) [reply usable_dataset_tar_extractor.py:59]: Neues ticket in Trello erstellt

**Pull request merged**